### PR TITLE
Allow callback as range in RangeValidator

### LINKED ISSUE #7581: RangeValidator: allow callback as range
In some situations will be useful to pass a callback to `range` param of `RangeValidator`.
Rule configuration array can be set like this:
```php
['city_id', 'in', 'range' => function ($model) use ($cities) { return $cities[$model->country_id]; }]
```
Now to pass value of another attribute of model (via `$this`) we need to use InlineValidator:
```php
['city_id', function ($attribute, $params) use ($cities) {
    if (!in_array($this->$attribute, $cities[$this->country_id])) {
        $this->addError($attribute, Yii::t('app', '{attribute} is not a valid city for specified country.', [
            'attribute' => $this->getAttributeLabel($attribute),
        ]));
    }
}],
```

### PR DESCRIPTION
**Pull request by RomeroMsk**
Fixes #7581

### CODE REVIEW

**Comment by samdark** [comment]
What's your use case?

File: framework/validators/RangeValidator.php [hunk 1]
```diff
@@ -23,7 +23,15 @@
 class RangeValidator extends Validator
 {
     /**
-     * @var array list of valid values that the attribute value should be among
+     * @var mixed list of valid values that the attribute value should be among or a PHP callable that returns
```
Comment (samdark) [inline framework/validators/RangeValidator.php]: PHP callable !== anonymous function

Comment (samdark) [inline framework/validators/RangeValidator.php]: Either fix description or fix method itself.

**Comment by samdark** [comment]
Looks good to me except closure !== valid PHP callback... @RomeroMsk was it meant to support all callback types or just closures?

File: framework/validators/RangeValidator.php (hunk 1 above)
Comment (RomeroMsk) [inline framework/validators/RangeValidator.php]: It was just copied from here:
https://github.com/yiisoft/yii2/blob/master/framework/validators/DefaultValueValidator.php#L22

Comment (samdark) [inline framework/validators/RangeValidator.php]: Then it's wrong in `DefaultValueValidator` as well.

**Comment by RomeroMsk** [comment]
Is it ok now?

**Comment by samdark** [comment]
Absolutely.

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/e93b797098338b349c5da76fd917fe7fd615943b. Thank you.

**Comment by RomeroMsk** [comment]
Thanks!

**Pull request closed without merging**